demisto_sdk/commands/content_graph/interface/neo4j_graph.py

```diff
@@ -0,0 +1,123 @@
+from pathlib import Path
+import neo4j
+from typing import Any, Dict, List, Optional, Tuple
+from demisto_sdk.commands.common.constants import MarketplaceVersions
+
+from demisto_sdk.commands.content_graph.common import ContentTypes, Rel
+from demisto_sdk.commands.content_graph.interface.graph import ContentGraphInterface
+from demisto_sdk.commands.content_graph.interface.neo4j.queries.indexes import create_indexes
+from demisto_sdk.commands.content_graph.interface.neo4j.queries.constraints import create_constraints
+from demisto_sdk.commands.content_graph.interface.neo4j.queries.nodes import (
+    create_nodes,
+    duplicates_exist,
+    get_packs_content_items,
+    get_all_integrations_with_commands,
+    delete_all_graph_nodes,
+    get_nodes_by_type,
+    search_nodes,
+)
+from demisto_sdk.commands.content_graph.interface.neo4j.queries.relationships import (
+    create_relationships,
+    get_relationships_by_type,
+)
+
+
+class Neo4jContentGraphInterface(ContentGraphInterface):
+    def __init__(self, database_uri, auth: Tuple[str, str]) -> None:
+        self.driver: neo4j.Neo4jDriver = neo4j.GraphDatabase.driver(database_uri, auth=auth)
+    
+    def create_indexes_and_constraints(self) -> None:
+        with self.driver.session() as session:
+            tx: neo4j.Transaction = session.begin_transaction()
+            create_indexes(tx)
+            create_constraints(tx)
+            tx.commit()
+            tx.close()
```

Comment:
What if an exception was raised in one of the methods (`create_indexes` or `create_constraints`)
Would the transaction be closed?
Shouldn't it be in a `finally` block?
Same for all other methods here.

Reply:
If there's an exception it won't commit (so no need to rollback). agree we need to put `close` under a finally block.